modin/engines/base/frame/data.py

```diff
@@ -234,6 +256,74 @@ def _filter_empties(self):
         self._column_widths_cache = [w for w in self._column_widths if w > 0]
         self._row_lengths_cache = [r for r in self._row_lengths if r > 0]
 
+    def _validate_axis_equality(self, axis: int, update: bool):
+        """
+        Validates internal and external indices of modin_frame at the specified axis.
+
+        Parameters
+        ----------
+            axis : int,
+                Axis to validate indices along
+            update : bool,
+                Update or raise an exception if indices don't match
+        """
+        internal_axis = self._frame_mgr_cls.get_indices(
+            axis, self._partitions, lambda df: df.axes[axis]
+        )
+        is_equals = self.axes[axis].equals(internal_axis)
+        if not update:
```

Comment:
I did not see this getting used in this PR, how are you expecting this to be used?

Reply:
It may be better to just leave the `update` parameter out because I don't think we ever want to throw the internal error. What do you think?

Reply:
well, if you think so

removed that parameter in the last commit